posts/REDCapTidieR_v0.3.qmd

```diff
@@ -0,0 +1,103 @@
+
+From the API, [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [events](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event) aren't captured much differently from [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [instruments](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#instrument). In fact, no new column names are added to the data export. The same default columns of `redcap_repeat_instrument` and `redcap_repeat_instance` are used to indicate a [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [instrument](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#instrument) when *both* are filled with data. It is the *lack* of data in `redcap_repeat_instrument` while data remains in `redcap_repeat_instance` which indicates a [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [event](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event). In the image below, a sample REDCap export is shown where the [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [instrument](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#instrument) data is shown in the [red box]{style="color: red;"} and the [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [*event*](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event) data is shown in the [green box]{style="color: green;"}:
+
+![](images/repeating_events_instrument_export.png){fig-alt="REDCap Repeating Event Export" fig-align="center" width="518"}
+
+> It is important to note that [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [instruments](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#instrument) **cannot** be included in [repeating](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#repeating) [events](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#event), and REDCapTidieR will throw an error if this is forced and detected!
+
+### What this means in REDCapTidieR
+
+For REDCapTidieR 0.3.0, we decided it is better to be explicit about what our [columns](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#data-column) mean. For this reason, `redcap_repeat_instance` has been split into `redcap_form_instance` and `redcap_event_instance` for applicable [databases](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#database). This is a **breaking change** in the output of the package.
+
+For the above sample database, the typical `REDCapTidieR::read_readcap()` output stays the same:
```

Comment:
I would specify we mean that the structure of the supertibble stays the same rather than the entire output